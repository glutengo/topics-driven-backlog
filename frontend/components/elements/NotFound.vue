<template>
    <section class="not-found">
        <h1>The page you're trying to view does not exist.</h1>

        <p>You can:</p>

        <ul>
            <li><router-back>Go back</router-back></li>
            <li><router-link to="/">Go to homepage</router-link></li>
        </ul>
    </section>
</template>

<script>
    import RouterBack from "./RouterBack.vue";

    export default {
        components: {RouterBack},
        name: 'NotFound'
    }
</script>