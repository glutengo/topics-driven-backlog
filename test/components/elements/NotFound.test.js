import NotFound from '@/components/elements/NotFound';
import { mount } from 'vue-test-utils';

describe('NotFound.test.js', () => {
    let cmp;

    beforeEach(() => {
        cmp = mount(NotFound);
    });

    it('is a section with the class not-found', () => {
        expect(cmp.is('section.not-found')).toBe(true);
    });

    it('has a link to go back', () => {
        const link = cmp.find('a');
        expect(link.exists()).toBe(true);
        expect(link.text()).toBe('Go back');
    });

    it('has a link to go to the homepage', () => {
        const routerLink = cmp.find('router-link');
        expect(routerLink.exists()).toBe(true);
        expect(routerLink.element.getAttribute('to')).toBe('/');
        expect(routerLink.text()).toBe('Go to homepage');
    });

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot();
    });
});
