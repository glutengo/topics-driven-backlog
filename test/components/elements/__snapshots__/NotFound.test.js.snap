// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.test.js has the expected html structure 1`] = `
<section
  class="not-found"
>
  <h1>
    The page you're trying to view does not exist.
  </h1>
   
  <p>
    You can:
  </p>
   
  <ul>
    <li>
      <a
        href="javascript:"
      >
        Go back
      </a>
    </li>
     
    <li>
      <router-link
        to="/"
      >
        Go to homepage
      </router-link>
    </li>
  </ul>
</section>
`;
